### PR TITLE
[WIP] Added documentation in readme and doc

### PR DESCRIPTION
**Pull request by shivamashtikar**
* added guideline for setting up custom keybindings
* added installation guid
* TODO: generate help tag so that :help can be used

### CODE REVIEW

**Comment by sriharshachilakapati** [comment]
Aren't help tags generated automatically? Pathogen generates it for all plugins if `HelpTags` command is added in the `.vimrc` file.

File: ftplugin/purescript_vimmerps.vim [hunk 1]
```diff
@@ -23,11 +23,11 @@ com! -buffer -nargs=1 Psearch call vimmerps#Psearch(<q-args>)
 
 if !g:vimmerps_disable_mappings
     " Keybindings for IDE like funtions
-    nm <buffer> <silent> <leader>a :Papply<CR>
-    nm <buffer> <silent> <leader>i :Pimport<CR>
-    nm <buffer> <silent> <leader>g :call LanguageClient_textDocument_definition()<CR>
-    nm <buffer> <silent> <leader>h :call LanguageClient_textDocument_hover()<CR>
-    nm <buffer> <silent> <leader>l :Pbuild<CR>
+   autocmd Filetype purescript nm <buffer> <silent> <leader>a :Papply<CR>
```
Comment (sriharshachilakapati) [inline ftplugin/purescript_vimmerps.vim]: This is not needed. This file is present in `ftplugin` and has `purescript_` in it's name. These won't be called in other buffers.

Comment (shivamashtikar) [reply ftplugin/purescript_vimmerps.vim]: yes @sriharshachilakapati you are right, the tag wasn't correct hence I was not able to use `help` function, now that I have updated tags, It'll work now

File: doc/vimmerps.txt [hunk 2]
```diff
@@ -1 +1,74 @@
...
+< Start purescript-language-server
+              *vimmer-ps-Pend*
+>
+  :Pend
+< Stop purescript-language-server
+              *vimmer-ps-Prestart*
+>
+  :Prestart
+< Restart purescript-language-server
+              *vimmer-ps-Prsearch*
```
Comment (sriharshachilakapati) [inline doc/vimmerps.txt]: Typo. It's `Psearch`.

**Review by sriharshachilakapati — CHANGES_REQUESTED** [review]

File: doc/vimmerps.txt (hunk 2 above)
Comment (shivamashtikar) [reply doc/vimmerps.txt]: resolved and squashed

**Review by sriharshachilakapati — APPROVED** [review]
Thanks!

**Pull request merged**